src/caja-desktop-window.c

```diff
@@ -252,34 +331,44 @@ unrealize (GtkWidget *widget)
 static void
 set_wmspec_desktop_hint (GdkWindow *window)
 {
-    GdkAtom atom;
+    GdkDisplay *display;
+    display = gdk_screen_get_display (gdk_screen_get_default());
+    if (GDK_IS_X11_DISPLAY (display))
+    {
+        GdkAtom atom;
 
-    atom = gdk_atom_intern ("_NET_WM_WINDOW_TYPE_DESKTOP", FALSE);
+        atom = gdk_atom_intern ("_NET_WM_WINDOW_TYPE_DESKTOP", FALSE);
 
-    gdk_property_change (window,
+        gdk_property_change (window,
                          gdk_atom_intern ("_NET_WM_WINDOW_TYPE", FALSE),
                          gdk_x11_xatom_to_atom (XA_ATOM), 32,
                          GDK_PROP_MODE_REPLACE, (guchar *) &atom, 1);
+    }
+
 }
 
 static void
 set_desktop_window_id (CajaDesktopWindow *window,
                        GdkWindow             *gdkwindow)
 {
-    /* Tuck the desktop windows xid in the root to indicate we own the desktop.
+    /* Tuck the desktop windows xid in the root to indicate we own the desktop in on x11
      */
-    Window window_xid;
-    GdkWindow *root_window;
+    GdkDisplay *display = gdk_screen_get_display (gdk_screen_get_default());
+    if (GDK_IS_X11_DISPLAY (display))
```

Comment:
Again, probably worth using
```suggestion
    if (GDK_IS_X11_WINDOW (gdkwindow))
```

Reply:
set_desktop_window_id  and set_wmspec_desktop_hint are now called only from inside  a 
`    if (GDK_IS_X11_DISPLAY (display))`
block in` realize`
unfortunately we had to use display there to get it to work, otherwise the x11 desktop sometimes rendered on top of other windows